opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1256,12 +1250,7 @@ namespace Opm {
 
             for (const auto& connection: well.getConnections())
             {
-                const size_t i = size_t(connection.getI());
-                const size_t j = size_t(connection.getJ());
-                const size_t k = size_t(connection.getK());
-
-                const size_t index = this->eclState_.gridDims().getGlobalIndex(i, j, k);
-                int compressed_idx = cartMap[index];
+                const int compressed_idx = compressedIndexForInterior(connection.global_index());
                 if ( compressed_idx >= 0 ) // Ignore completions in inactive/remote cells.
                 {
                     compressed_well_perforations.push_back(compressed_idx);
```

Comment:
Completely unrelated: Why is this vector sorted afterwards? Does seem to buy us anything in the way it is used.

Reply:
It may not be significant, but it could perhaps make the access order more predictable when accessing the matrix later?